test/cellarea.jl

```diff
@@ -0,0 +1,59 @@
+using Rasters, DimensionalData, Rasters.Lookups, ArchGDAL
+using Test
+using DimensionalData: @dim, YDim
+include(joinpath(dirname(pathof(Rasters)), "../test/test_utils.jl"))
+
+@testset "cellarea" begin
+    x = X(Projected(90.0:0.1:99.9; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(0.1), crs=EPSG(4326), dim = X()))
+    y = Y(Projected(0.0:0.1:89.9; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(0.1), crs=EPSG(4326), dim = Y()))
+    y_rev = Y(Projected(89.9:-0.1:0; sampling=Intervals(Start()), order = ReverseOrdered(), span = Regular(-0.1), crs=EPSG(4326), dim = Y()))
+    dimz = (x,y)
+
+    dimz_25832 = X(Projected(0.0:100:10000.0; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(100), crs=EPSG(25832))),
+       Y(Projected(0.0:100:10000.0; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(100), crs=EPSG(25832)))
+    ras = ones(dimz)
+
+    cs = cellarea(dimz)
+    cs2 = cellarea(dimz_25832)
+    cs3 = cellarea((x, y_rev))
+
+    # check cellsize still works
+    csold = @test_warn "deprecated" cellsize(dimz)
+    @test all(csold .≈ (cs ./ 1e6))
+
+    # Check the output is a raster 
+    @test cs isa Raster
+    @test cs2 isa Raster
+    # Test that the total area matches the expected area (1/72th of the Earth surface)
+    @test sum(cs) ≈ sum(cs3)
+    @test sum(cs) ≈ 510.1e12/72 rtol = 0.01
+    # Test all areas are about 10,000 m2
+    @test maximum(cs2) ≈ 1e4 rtol = 0.01
+    @test minimum(cs2) ≈ 1e4 rtol = 0.01
+    # test passing in a raster or dims gives the same result
+    cs_ras = cellarea(ras)
+    @test cs == cs_ras
+
+    # test a YDim other than Y is handled correctly
+    @dim Lat YDim "latitude"
+    lat = Lat(Projected(0.0:0.1:89.9; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(0.1), crs=EPSG(4326)))
+    cs_lat = cellarea((x, lat))
+    @test parent(cs_lat) == parent(cs)
+    @test dims(cs_lat) == (x, lat)
+
+    # test using mapped dims to calculate cellsize
+    x_3857 = X(Projected(0:1e5:1e7; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(1e5), crs=EPSG(3857), dim = X()))
+    y_3857 = Y(Projected(0:1e5:1e7; sampling=Intervals(Start()), order = ForwardOrdered(), span = Regular(1e5), crs=EPSG(3857), dim = Y()))
+    dimz_3857 = (x_3857, y_3857)
+    mappeddimz = setmappedcrs(dimz_3857, EPSG(4326))
```

Comment:
I think it's most commonly the other way around?  

Where you have a CRS that _can_ be remapped to lat/long, we should check if it can be done and then compute cell area through the fast path.

Reply:
With "the other way around" do you mean cases where the crs is lat/lon and the mappedcrs is something different? In that case cellarea will use the crs.

Reply:
cellarea will only use mappedcrs if the crs is not in degrees, but mappedcrs is, so that's why the test is this way around.